packages/@aws-cdk/aws-ec2-alpha/README.md

```diff
@@ -366,3 +362,104 @@ myVpc.addInternetGateway({
   ipv4Destination: '192.168.0.0/16',
 });
 ```
+
+## Importing an existing VPC
+
+You can import an existing VPC and its subnets using the `VpcV2.fromVpcV2Attributes()` method or an individual subnet using `SubnetV2.fromSubnetV2Attributes()` method.
+
+### Importing a VPC
+
+To import an existing VPC, use the `VpcV2.fromVpcV2Attributes()` method. You'll need to provide the VPC ID, primary CIDR block, and information about the subnets. You can import secondary address as well created through IPAM, BYOIP(IPv4) or enabled through Amazon Provided IPv6. You must provide VPC Id and its primary CIDR block for importing it.
+
+If you wish to add a new subnet to imported VPC, new subnet's IP range(IPv4) will be validated against provided secondary and primary address block to confirm that it is within the the range of VPC.
+
+Here's an example of how to import a VPC with multiple CIDR blocks, IPv6 support, and different subnet types:
+
+In this example, we're importing a VPC with:
+
+ - A primary CIDR block (10.1.0.0/16)
+ - One secondary IPv4 CIDR block (10.2.0.0/16)
+ - Two secondary address using IPAM pool (IPv4 and IPv6)
+ - VPC has Amazon-provided IPv6 CIDR enabled
+ - An isolated subnet in us-west-2a
+ - A public subnet in us-west-2b
+
+```ts
+
+const stack = new Stack();
+
+const importedVpc = VpcV2.fromVpcV2Attributes(this, 'ImportedVPC', {
+  vpcId: 'vpc-XXX',
+  vpcCidrBlock: '10.1.0.0/16',
+  secondaryCidrBlocks: [
+    {
+      cidrBlock: '10.2.0.0/16',
+      cidrBlockName: 'ImportedBlock1',
+    },
+    {
+      ipv6IpamPoolId: 'ipam-pool-XXX',
+      ipv6NetmaskLength: 52,
+      cidrBlockName: 'ImportedIpamIpv6',
+    },
+    {
+      ipv4IpamPoolId: 'ipam-pool-XXX',
+      ipv4IpamProvisionedCidrs: ['10.2.0.0/16'],
+      cidrBlockName: 'ImportedIpamIpv4',
+    },
+    {
+      amazonProvidedIpv6CidrBlock: true,
+    }
+  ],
+  isolatedSubnets: [{
+    subnetName: 'IsolatedSubnet2',
+    subnetId: 'subnet-03cd773c0fe08ed26',
+    subnetType: SubnetType.PRIVATE_ISOLATED,
+    availabilityZone: 'us-west-2a',
+    ipv4CidrBlock: '10.2.0.0/24',
+    routeTableId: 'rtb-0871c310f98da2cbb',
```

Comment:
what is the value of providing the route table id? if it is needed, is it better to provide the route table object ?

Reply:
it will be used to add routes when attaching gateways, since routeTable interface only has the routeTableId as the property, i think just providing the Id should be enough to modify the class field using just the id     ```
this.routeTable = {
      routeTableId: props.routeTableId!,
    };
```

Reply:
honestly, I do not have a strong opinion about it. The only thing I have in mind is passing an object will give us and customers other options to create the route table, and do more logic by invoking other functions.

In general in CDK, we prefer using strong typed Objects more than Ids.